### PR TITLE
rpc: Reject deprecated reserveChangeKey in fundrawtransaction

### PR DESCRIPTION
**Pull request by maflcko**

### CODE REVIEW

**Review by randolf — CHANGES_REQUESTED** [review]
Please change "rpc" to "RPC" because it's an acronym, thanks.
File:  doc/release-notes.md

File: doc/release-notes.md [hunk 1]
```diff
@@ -56,6 +56,13 @@ frequently tested on them.
 Notable changes
 ===============
 
+RPC changes
+------------
```
Comment (promag) [inline doc/release-notes.md]: Remove extra `-`.

File: src/wallet/rpcwallet.cpp [hunk 2]
```diff
@@ -3129,7 +3129,6 @@ UniValue fundrawtransaction(const JSONRPCRequest& request)
                 {"change_type", UniValueType(UniValue::VSTR)},
                 {"includeWatching", UniValueType(UniValue::VBOOL)},
                 {"lockUnspents", UniValueType(UniValue::VBOOL)},
-                {"reserveChangeKey", UniValueType(UniValue::VBOOL)}, // DEPRECATED (and ignored), should be removed in 0.16 or so.
```
Comment (promag) [inline src/wallet/rpcwallet.cpp]: Should we allow it behind `IsDeprecatedRPCEnabled()`?

Comment (jonasschnelli) [reply src/wallet/rpcwallet.cpp]: I guess this move is from deprecated to removed. So IMO no need to hide it behind `IsDeprecatedRPCEnabled()`.

**Comment by jonasschnelli** [comment]
utACK fa5f51830d2649cace2fc827084093019497c151

**Comment by laanwj** [comment]
utACK https://github.com/bitcoin/bitcoin/commit/fa5f51830d2649cace2fc827084093019497c151, strange that we missed this in #12336.

**Comment by promag** [comment]
> utACK fa5f518, strange that we missed this in #12336.

@laanwj it wasn't https://github.com/bitcoin/bitcoin/pull/12336#pullrequestreview-95040479.

**Comment by laanwj** [comment]
> @laanwj it wasn't #12336 (review).

Oh I missed that, sorry!

**Pull request merged**